src/t8_geometry/t8_geometry_base.hxx

```diff
@@ -33,14 +33,24 @@
 #include <t8_forest/t8_forest.h>
 #include <t8_geometry/t8_geometry.h>
 
+#include <string>
+#include <functional>
+
 T8_EXTERN_C_BEGIN ();
 
+/**
+ * The base class for all geometries.
+ * This class provides a general template for all geometries.
+ * It is a pure virtual class and has to be inherited by a concrete
+ * geometry implementation.
+ */
 struct t8_geometry
 {
  public:
   /* Basic constructor that sets the dimension, the name, and the name for the attribute. */
```

Comment:
```suggestion
  /* Basic constructor that sets the dimension, the name/hash, and the name for the attribute. */
```

Reply:
I would not name the hash explicitly, since it is not exposed as an argument